src/hdmf/utils.py

```diff
@@ -954,6 +954,21 @@ def to_uint_array(arr):
     raise ValueError('Cannot convert array of dtype %s to uint.' % arr.dtype)
 
 
+def is_ragged(data):
+    """
+    Test whether a list of lists or array is ragged / jagged
+    """
+    if hasattr(data, '__len__') and not isinstance(data, str):
```

Comment:
I think this check may also need a ``and not instance(data, AbstractDataChunkInterator)``

Reply:
If we limit the check to only lists/tuples I think that should cover this